### PR TITLE
Update Collection Requirements

### PR DESCRIPTION
**Pull request by Andersson007**
##### SUMMARY
Update Collection Requirements

### CODE REVIEW

**Comment by Andersson007** [comment]
ready for review

File: collection_requirements.rst [hunk 1]
```diff
@@ -385,6 +386,5 @@ To be included in the `ansible` package, collections must meet the following cri
 Other things
 ============
 
-* ansible-base's runtime.yml
-* After content is (moved out of community.general or community.network) OR new collection satisfies all the requirements
-    * Add the collection to the ``ansible.in`` file in a corresponding directory of `ansible-build-data repository <https://github.com/ansible-community/ansible-build-data/>`_
+* ansible-core's runtime.yml
```
Comment (Andersson007) [inline collection_requirements.rst:389]:
```suggestion
* ansible-core's runtime.yml
```
Can we remove this line?

File: collection_requirements.rst [hunk 2]
```diff
@@ -1,73 +1,76 @@
...
 
-Be sure you're subscribed to:
-
-* `Changes impacting Collections <https://github.com/ansible-collections/overview/issues/45>`_ to track changes that Collection maintainers should be aware of
-* The Bullhorn, a newsletter for the Ansible developer community, `back issues and how to add content <https://github.com/ansible/community/issues/546>`_
+Be sure you are subscribed to:
 
-Why is this needed
-===================
+* The `news-for-maintainers repository <https://github.com/ansible-collections/news-for-maintainers>`_ to track changes that Collection maintainers should be aware of.
```
Comment (felixfontein) [inline collection_requirements.rst:32]:
```suggestion
* The `news-for-maintainers repository <https://github.com/ansible-collections/news-for-maintainers>`_ to track changes that Collection maintainers should be aware of. Subscribe only to issues if you want less traffic.
```

File: collection_requirements.rst [hunk 3]
```diff
@@ -1,73 +1,76 @@
...
+* The `Bullhorn <https://github.com/ansible/community/wiki/News#the-bullhorn>`_ Ansible contributor newsletter.
 
 Collection Infrastructure
 =========================
 
-* MUST have a publicly available issue tracker, that does not require a paid level of service to create an account or view issues.
-* Collections MUST have a Code of Conduct (CoC)
+Community collections
```
Comment (felixfontein) [inline collection_requirements.rst:38]: Is this about `community.*` collections, or about collections included in the Ansible package? The below rules are about the latter, but this sounds like the former. How about:
```suggestion
Collections included in the Ansible community package
```

File: collection_requirements.rst [hunk 4]
```diff
@@ -1,73 +1,76 @@
...
+A collection MUST be developed and tested using the below Python requirements as Ansible supports a wide variety of machines.
+
+The collection should adhere to the tips mentioned in the official `Ansible Development Guide <https://docs.ansible.com/ansible/latest/dev_guide/developing_python_3.html#ansible-and-python-3>`_.
 
 Python Requirements
 -------------------
 
-Python requirements for a collection varies between controller-environment and other-environment. On the controller-environment, the Python versions required may be higher than what is required on the other-environment. While developing a collection, you need to understand the definitions of both  controller-environment and other-environment to help you choose Python versions accordingly: 
+Python requirements for a collection vary between ``controller-environment`` and ``other-environment``. On the controller-environment, the Python versions required may be higher than what is required on the other-environment. While developing a collection, you need to understand the definitions of both  controller-environment and other-environment to help you choose Python versions accordingly:
```
Comment (felixfontein) [inline collection_requirements.rst:60]: Or maybe also single `*`. But double backticks is the wrong formatting IMO, except if we always use teletype for these words (which we do not).
```suggestion
Python requirements for a collection vary between **controller-environment** and **other-environment**. On the controller-environment, the Python versions required may be higher than what is required on the other-environment. While developing a collection, you need to understand the definitions of both  controller-environment and other-environment to help you choose Python versions accordingly: 
```

File: collection_requirements.rst [hunk 5]
```diff
@@ -1,73 +1,76 @@
...
+Python requirements for a collection vary between ``controller-environment`` and ``other-environment``. On the controller-environment, the Python versions required may be higher than what is required on the other-environment. While developing a collection, you need to understand the definitions of both  controller-environment and other-environment to help you choose Python versions accordingly: 
 
-- ``controller-environment``: The plugins/modules always run in the same environment (Python interpreter, venv, host, etc) as ansible-core itself.
-- ``other-environment``: It is possible, even if uncommon in practice, for the plugins/modules to run in a different environment than ansible-core itself.
+* ``controller-environment``: The plugins/modules always run in the same environment (Python interpreter, venv, host, etc) as ansible-core itself.
+* ``other-environment``: It is possible, even if uncommon in practice, for the plugins/modules to run in a different environment than ansible-core itself.
```
Comment (felixfontein) [inline collection_requirements.rst:63]: Same here.

File: collection_requirements.rst [hunk 6]
```diff
@@ -1,73 +1,76 @@
...
 
 One example scenario where the ``even if`` clause comes into play is when using Cloud modules. These modules mostly run on the controller node but in some environments, the controller might run on one machine inside a demilitarized zone which cannot directly access the cloud machines. The user has to have the cloud modules run on a bastion host/jump server which has access to the cloud machines.
```
Comment (felixfontein) [inline collection_requirements.rst:65]:
```suggestion
One example scenario where the "even if" clause comes into play is when using Cloud modules. These modules mostly run on the controller node but in some environments, the controller might run on one machine inside a demilitarized zone which cannot directly access the cloud machines. The user has to have the cloud modules run on a bastion host/jump server which has access to the cloud machines.
```

File: collection_requirements.rst [hunk 7]
```diff
@@ -157,9 +158,9 @@ Documentation
...
 
-* Include a `DOCUMENTATION <https://docs.ansible.com/ansible/devel/dev_guide/developing_modules_documenting.html#documentation-block>`_ block
-* Include an `EXAMPLES <https://docs.ansible.com/ansible/devel/dev_guide/developing_modules_documenting.html#examples-block>`_ block (except where not relevant for the plugin type)
-* Use FQCNs when referring to modules, plugins and documentation fragments inside and outside the collection (including ``ansible.builtin.`` for the listed entities from `Ansible-base <https://docs.ansible.com/ansible/devel/collections/ansible/builtin/>`_)
+* Include a `DOCUMENTATION <https://docs.ansible.com/ansible/devel/dev_guide/developing_modules_documenting.html#documentation-block>`_ block.
+* Include an `EXAMPLES <https://docs.ansible.com/ansible/devel/dev_guide/developing_modules_documenting.html#examples-block>`_ block (except where not relevant for the plugin type).
+* Use FQCNs when referring to modules, plugins and documentation fragments inside and outside the collection (including ``ansible.builtin.`` for the listed entities from `Ansible-base <https://docs.ansible.com/ansible/devel/collections/ansible/builtin/>`_).
```
Comment (felixfontein) [inline collection_requirements.rst:163]:
```suggestion
* Use FQCNs when referring to modules, plugins and documentation fragments inside and outside the collection (including ``ansible.builtin.`` for the listed entities from `Ansible-core <https://docs.ansible.com/ansible/devel/collections/ansible/builtin/>`_).
```

File: collection_requirements.rst [hunk 8]
```diff
@@ -181,10 +182,9 @@ Changelogs
 
 Collections are required to include a changelog.  To give a consistent feel for changelogs across collections and ensure changelogs exist for collections included in the ``ansible`` package we suggest you use `antsibull-changelog <https://github.com/ansible-community/antsibull-changelog>`_ to maintain and generate this but other options exist.  Preferred (in descending order):
 
-
-1. Use antsibull-changelog (preferred)
-2. Provide ``changelogs/changelog.yaml`` in the `correct format <https://github.com/ansible-community/antsibull-changelog/blob/main/docs/changelog.yaml-format.md>`_
-3. Provide a link to the changelog file (self-hosted) (not recommended)
+#. Use antsibull-changelog (preferred).
+#. Provide ``changelogs/changelog.yaml`` in the `correct format <https://github.com/ansible-community/antsibull-changelog/blob/main/docs/changelog.yaml-format.md>`_.
```
Comment (felixfontein) [inline collection_requirements.rst:186]:
```suggestion
#. Provide ``changelogs/changelog.yaml`` in the `correct format <https://github.com/ansible-community/antsibull-changelog/blob/main/docs/changelog.yaml-format.md>`_. (You can use ``antsibull-lint changelog-yaml /path/to/changelog.yaml`` to validate the format.)
```

File: collection_requirements.rst [hunk 9]
```diff
@@ -267,11 +267,10 @@ allowed.
 These guidelines are the policy for inclusion in the Ansible package and are in addition to any
 licensing and legal concerns that may otherwise affect your code.
 
-
 Repository management
 =====================
 
-Every collection MUST have a public SCM repository, and releases of the collection MUST be tagged in this repository.
+Every Community collection MUST have a public SCM repository, and releases of the collection MUST be tagged in this repository.
```
Comment (felixfontein) [inline collection_requirements.rst:273]:
```suggestion
Every collection MUST have a public SCM repository, and releases of the collection MUST be tagged in this repository.
```
Same here. This applies to **all** collections included in Ansible, not only to `community.*` collections.

File: collection_requirements.rst [hunk 10]
```diff
@@ -319,42 +320,42 @@ CI Testing
   * All entries in ignores.txt MUST have a justification in a comment in the ignore.txt file for each entry.  For example ``plugins/modules/docker_container.py use-argspec-type-path # uses colon-separated paths, can't use type=path``.
   * Reviewers can block acceptance of a new collection if they don't agree with the ignores.txt entries.
 
-* You MUST run CI against each of the "major versions" (2.10, 2.11, 2.12, etc) of ``ansible-base``/``ansible-core`` that the collection supports. (Usually the ``HEAD`` of the stable-xxx branches.)
+* You MUST run CI against each of the ``major versions`` (2.10, 2.11, 2.12, etc) of ``ansible-base``/``ansible-core`` that the collection supports. (Usually the ``HEAD`` of the stable-xxx branches.)
```
Comment (felixfontein) [inline collection_requirements.rst:323]: Double backticks is not a substitute for quotation marks.
```suggestion
* You MUST run CI against each of the "major versions" (2.10, 2.11, 2.12, etc) of ``ansible-base``/``ansible-core`` that the collection supports. (Usually the ``HEAD`` of the stable-xxx branches.)
```

File: collection_requirements.rst (hunk 1 above)
Comment (felixfontein) [reply collection_requirements.rst:389]: I think so, since changes to it are no longer accepted AFAIK.

File: collection_requirements.rst (hunk 2 above)
Comment (Andersson007) [reply collection_requirements.rst:32]: makes sense to mention:)

File: collection_requirements.rst (hunk 9 above)
Comment (Andersson007) [reply collection_requirements.rst:273]: `Community` is not about `community.*` collections. Its aim is to highlight that this is not about collections living on Automation Hub

File: collection_requirements.rst (hunk 3 above)
Comment (Andersson007) [reply collection_requirements.rst:38]: Works for me:)

File: collection_requirements.rst (hunk 5 above)
Comment (Andersson007) [reply collection_requirements.rst:63]: i think the bullet points highlight them enough without any additional formatting, so I'll just remove it

File: collection_requirements.rst (hunk 1 above)
Comment (Andersson007) [reply collection_requirements.rst]: Removed, thanks

File: collection_requirements.rst [hunk 11]
```diff
@@ -1,73 +1,76 @@
-*****************************
-Ansible Collections Checklist
-*****************************
+******************************************
+Ansible Community Collections Requirements
```
Comment (felixfontein) [inline collection_requirements.rst:2]: To emphasize that this is not for Community Collections, but for collections included in the Ansible Community Package:
```suggestion
Ansible Community Package Collections Requirements
```

File: collection_requirements.rst (hunk 9 above)
Comment (Andersson007) [reply collection_requirements.rst]: Solved

File: collection_requirements.rst (hunk 11 above)
Comment (Andersson007) [reply collection_requirements.rst:2]: Sounds good, committed, thanks!

File: collection_requirements.rst [hunk 12]
```diff
@@ -1,73 +1,76 @@
...
 
 Feedback
 ========
 
-As with any project it's very important that we get feedback from users, contributors and maintainers. We recognize that the move to Collections is a big change in how Ansible is developed and delivered.
+As with any project it's very important that we get feedback from users, contributors, and maintainers. We recognize that the move to Collections is a big change in how Ansible is developed and delivered.
 
 Please raise feedback by:
 
-* Discussing in the ``#ansible-community`` `libera.chat <https://libera.chat/>`_ IRC channel.
```
[the remark this answers is not in the document]
Comment (Andersson007) [reply collection_requirements.rst:23]: Changed, PTAL

File: collection_requirements.rst [hunk 13]
```diff
@@ -1,73 +1,77 @@
...
-* Adding to the `Community Working Group IRC meeting <https://github.com/ansible/community/issues/539>`_.
-* Creating `GitHub Issues <https://github.com/ansible-collections/overview/issues>`_ against this repository.
+* Discussing in the `#community:ansible.com <https://matrix.to/#/#community:ansible.com>`_ Matrix channel bridget with the ``#ansible-community`` channel on Libera.Chat IRC. See the `Ansible Communication Guide <https://docs.ansible.com/ansible/latest/community/communication.html#real-time-chat>`_ for details.
```
Comment (felixfontein) [inline collection_requirements.rst:23]:
```suggestion
* Discussing in the `#community:ansible.com Matrix room <https://matrix.to/#/#community:ansible.com>`_ bridges with the ``#ansible-community`` channel on Libera.Chat IRC. See the `Ansible Communication Guide <https://docs.ansible.com/ansible/latest/community/communication.html#real-time-chat>`_ for details.
```

File: collection_requirements.rst [hunk 14]
```diff
@@ -294,9 +293,11 @@ Branch protections MUST be enforced:
 CI Testing
 ===========
```
[the remark this answers is not in the document]
Comment (Andersson007) [reply collection_requirements.rst:296]: OK, will think of this a bit later

[the remark this answers is not in the document]
Comment (Andersson007) [reply collection_requirements.rst:296]: added, PTAL

**Review by gundalow — APPROVED** [review]
Massive improvement over what we have.
Thanks for all your work on this.
@felixfontein I really appreciate you helping _with_ this.

File: collection_requirements.rst [hunk 15]
```diff
@@ -1,73 +1,77 @@
...
+**************************************************
 
 .. contents:: Topics
 
 Overview
 ========
-This document is for maintainers of collections to provide them help, advice, and guidance on making sure their collections are correct.
 
-**Warning:** Subject to frequent updates. This is a "living document", expect it to change as we progress with the Collections work over the next few months.
+This document is for maintainers of Ansible Community collections, living in the `ansible-collections <https://github.com/ansible-collections>`_ repository or included in the Ansible community package (hereinafter the Collections), to provide them help, advice, and guidance on making sure their collections are correct.
```
Comment (felixfontein) [inline collection_requirements.rst:10]:
```suggestion
This document is for maintainers of Ansible Community collections, living in the `ansible-collections <https://github.com/ansible-collections>`_ repository or included in the Ansible community package (hereinafter the collections), to provide them help, advice, and guidance on making sure their collections are correct.
```
Do we really have to use a capital C here? If we do, there are a LOT of more places to check and possibly adjust (I saw a few which would need a capital C as well). I think it's better to stick to the lower-case form.

Comment (Andersson007) [reply collection_requirements.rst:10]: yep, let's use it in lower-case

Comment (Andersson007) [reply collection_requirements.rst]: changed everywhere, thanks!

**Comment by Andersson007** [comment]
@gundalow @felixfontein thanks for reviewing!

**Pull request merged**